neuronpp/utils/record.py

```diff
@@ -143,17 +151,66 @@ def _plot_animate(self, steps=10000, y_lim=None, position=None):
 
                 ax.set_xlim(t.min(), t.max())
                 if y_lim is None:
-                    ax.set_ylim(r.min()-(np.abs(r.min()*0.05)), r.max()+(np.abs(r.max()*0.05)))
+                    y_limits = (r.min() - (np.abs(r.min() * 0.05)), r.max() + (np.abs(r.max() * 0.05)))
+                    ax.set_ylim(y_limits)
 
                 # update data
                 line.set_data(t, r)
-
+                if show_true_predicted:
+                    # info draw triangles for true and predicted classes
+                    if true_labels is not None:
+                        true_x, pred_x = self._true_predicted_class_marks(label=true_labels[i], true_class=true_class,
+                                                                          pred_class=pred_class, t=t, r=r,
+                                                                          stepsize=stepsize, dt=dt)
+                    else:
+                        raise ValueError("True_labels parameter need to be given if show_true_prediction is True")
+                    if y_lim is None:
+                        true_y = [y_limits[0] + np.abs(y_limits[0]) * 0.09] * len(true_x)
+                        pred_y = [y_limits[1] - np.abs(y_limits[1] * 0.11)] * len(pred_x)
+                    else:
+                        true_y = [y_lim[0]] * len(true_x)
+                        pred_y = [y_lim[1]] * len(pred_x)
+                    ax.scatter(true_x, true_y, c="orange", marker="^", alpha=0.95, label="true")
+                    ax.scatter(pred_x, pred_y, c="magenta", marker="v", alpha=0.95, label="predicted")
+                    if create_fig and i == 0:
+                        # draw legend only the first time and only on the uppermost graph
+                        ax.legend()
+
+            # info join plots by removing labels and ticks from subplots that are not on the edge
+            if create_fig:
```

Comment:
Shouldn't this part be places under the condition `if show_true_predicted` as well? Because it is relevant only for marks

Reply:
No, it is relevant for all the situations: it simply joins the plots which have the **same** OX values. The space is wasted, as well as the tick marks, xlabel which in the basic form usually is overlapping, etc. Generally, it looked ugly, sorry to say.